packages/examples/cvat/exchange-oracle/src/handlers/job_creation.py

```diff
@@ -326,42 +388,127 @@ def build(self):
             db_service.get_project_by_id(session, project_id, for_update=True)  # lock the row
             db_service.add_project_images(session, cvat_project.id, data_filenames)
 
-            for job_filenames in job_configuration:
-                cvat_task = cvat_api.create_task(cvat_project.id, escrow_address)
-
+            for data_subset in self._split_dataset_per_task(
+                data_to_be_annotated,
+                subset_size=Config.cvat_config.cvat_max_jobs_per_task * segment_size,
+            ):
+                cvat_task = cvat_api.create_task(
+                    cvat_project.id, escrow_address, segment_size=segment_size
+                )
                 task_id = db_service.create_task(
                     session, cvat_task.id, cvat_project.id, TaskStatuses[cvat_task.status]
                 )
                 db_service.get_task_by_id(session, task_id, for_update=True)  # lock the row
 
-                # Actual task creation in CVAT takes some time, so it's done in an async process.
-                # The task is fully created once 'update:task' or 'update:job' webhook is received.
+                # The task is fully created once 'update:task' webhook is received.
                 cvat_api.put_task_data(
                     cvat_task.id,
                     cloud_storage.id,
-                    filenames=job_filenames,
+                    filenames=data_subset,
                     sort_images=False,
+                    # use the same value for the chunk size as for the job size
+                    chunk_size=segment_size,
```

Comment:
Need to include validation frames for chunk sizes

Reply:
Do we need to add an upper limit for the `chunk_size`?

Reply:
No, I don't think so. The parameter should be removed or redesigned later